sam/src/main/kotlin/org/taktik/icure/services/external/rest/v1/controllers/be/SamV2Controller.kt

```diff
@@ -69,551 +71,545 @@ import java.util.*
-            n.copy(productId = (if (index < productIds.size) productIds[index]?.takeIf { it.id == "SAMID:${n.id}" }?.productId else null) ?: productIds.find { it?.id == "SAMID:${n.id}" }?.productId)
-        }
-    }
-
-    private fun addProductIdsToAmps(amps: Flow<AmpDto>) = flow {
-        val acc = ArrayDeque<AmpDto>(20)
-        amps.collect {
-            acc.add(it)
-            if (acc.size == 20) {
-                addProductIdsToAmps(acc).forEach { emit(it) }
-                acc.clear()
-            }
-        }
-        addProductIdsToAmps(acc).forEach { emit(it) }
-    }
-
-    private fun addProductIdsToVmpGroups(vmpGroups: Flow<VmpGroupDto>) = flow {
-        val acc = ArrayDeque<VmpGroupDto>(20)
-        vmpGroups.collect {
-            acc.add(it)
-            if (acc.size == 20) {
-                addProductIdsToVmpGroups(acc).forEach { emit(it) }
-                acc.clear()
-            }
-        }
-        addProductIdsToVmpGroups(acc).forEach { emit(it) }
-    }
-
-    private fun addProductIdsToNmps(nmps: Flow<NmpDto>) = flow {
-        val acc = ArrayDeque<NmpDto>(20)
-        nmps.collect {
-            acc.add(it)
-            if (acc.size == 20) {
-                addProductIdsToNmps(acc).forEach { emit(it) }
-                acc.clear()
-            }
-        }
-        addProductIdsToNmps(acc).forEach { emit(it) }
-    }
+	companion object {
+		const val DEFAULT_LIMIT = 1000
+	}
+
+	private val proxyWebClient = WebClient.builder().build()
+
+	@Operation(summary = "Get Samv2 version.", description = "Returns a list of codes matched with given input. If several types are provided, paginantion is not supported")
+	@GetMapping("/v")
+	fun getSamVersion() = mono { samV2Logic.getVersion()?.let { samVersionMapper.map(it) } }
+
+	@Operation(summary = "Finding AMPs by label with pagination.", description = "Returns a list of codes matched with given input. If several types are provided, paginantion is not supported")
+	@GetMapping("/amp")
+	fun findPaginatedAmpsByLabel(
+		@Parameter(description = "language")
+		@RequestParam(required = false)
+		language: String?,
+		@Parameter(description = "label")
+		@RequestParam(required = false)
+		label: String?,
+		@Parameter(description = "The start key for pagination: a JSON representation of an array containing all the necessary components to form the Complex Key's startKey")
+		@RequestParam(required = false)
+		startKey: String?,
+		@Parameter(description = "An amp document ID")
+		@RequestParam(required = false)
+		startDocumentId: String?,
+		@Parameter(description = "Number of rows")
+		@RequestParam(required = false)
+		limit: Int?,
+	): PaginatedFlux<AmpDto> {
+		if (label == null || label.trim().length < 3) {
+			throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Label must be at least 3 characters long")
+		}
+
+		val realLimit = limit ?: SamV2Controller.DEFAULT_LIMIT
+		val paginationOffset = PaginationOffset(null, startDocumentId, null, realLimit)
```

Comment:
No pagination no cry

Reply:
The start key is not used in the logic, that's why I set it to null